spring-cloud-aws-sqs/src/test/java/io/awspring/cloud/sqs/listener/errorhandler/ImmediateRetryAsyncErrorHandlerTests.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2013-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs.listener.errorhandler;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+import io.awspring.cloud.sqs.listener.BatchVisibility;
+import io.awspring.cloud.sqs.listener.QueueMessageVisibility;
+import io.awspring.cloud.sqs.listener.SqsHeaders;
+import io.awspring.cloud.sqs.listener.Visibility;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+
+/**
+ * Tests for {@link ImmediateRetryAsyncErrorHandler}.
+ *
+ * @author Bruno Garcia
+ * @author Rafael Pavarini
+ */
+@SuppressWarnings("unchecked")
+class ImmediateRetryAsyncErrorHandlerTests {
+
+	@Test
+	void shouldChangeVisibilityToZero() {
+		Message<Object> message = mock(Message.class);
+		RuntimeException exception = new RuntimeException("Expected exception from shouldChangeVisibilityToZero");
+		MessageHeaders headers = mock(MessageHeaders.class);
+		Visibility visibility = mock(Visibility.class);
+		when(message.getHeaders()).thenReturn(headers);
```

Comment:
Would you mind changing when / verify to Mockito's BDD `given().willReturn()` / `then().should()`?